src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -34,17 +35,35 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
 
         public virtual async Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingContext bindingContext)
         {
-            var newBindingContext = CreateNewBindingContext(bindingContext,
-                                                            bindingContext.ModelName);
-
+            var newBindingContext = CreateNewBindingContext(bindingContext, bindingContext.ModelName);
             var modelBindingResult = await TryBind(newBindingContext);
-            if (modelBindingResult == null && !string.IsNullOrEmpty(bindingContext.ModelName)
-                && bindingContext.FallbackToEmptyPrefix)
+
+            if (bindingContext.FallbackToEmptyPrefix && !string.IsNullOrEmpty(bindingContext.ModelName))
             {
-                // fallback to empty prefix?
-                newBindingContext = CreateNewBindingContext(bindingContext,
-                                                            modelName: string.Empty);
-                modelBindingResult = await TryBind(newBindingContext);
+                var fallback = modelBindingResult == null;
+                if (!fallback && modelBindingResult.IsModelSet && modelBindingResult.IsEmptyModel)
+                {
+                    // Special case the empty object (i.e. contains at most default values) MutableObjectModelBinder
+                    // may return.
+                    fallback = true;
+
+                    // Remove any validation errors from first attempt; these are likely Required errors.
+                    // Clear entries completely, unlike ModelState.ClearValidationState().
+                    var entries =
+                        DictionaryHelper.FindKeysWithPrefix(bindingContext.ModelState, bindingContext.ModelName);
+                    foreach (var entry in entries.ToList())
+                    {
+                        bindingContext.ModelState.Remove(entry.Key);
+                    }
```

Comment:
I suggest the underlying issue is less the design than the apparent requirement to keep filling in empty objects when nothing from the value providers match.  see for example the closed issue #1488.  `MutableObjectModelBinder` is thus going to be a strange beast regardless of the surrounding architecture.